pkg/distributor/distributor.go

```diff
@@ -250,6 +251,11 @@ func New(cfg Config, clientConfig ingester_client.Config, limits *validation.Ove
 			Name:      "distributor_received_samples_total",
 			Help:      "The total number of received samples, excluding rejected and deduped samples.",
 		}, []string{"user"}),
+		discardedSamples: promauto.With(reg).NewCounterVec(prometheus.CounterOpts{
```

Comment:
I think we should re-use `cortex_discarded_samples_total` defined at https://github.com/cortexproject/cortex/blob/34bb142b129bd781539001fb1629cf2f3ed54b82/pkg/util/validation/validate.go#L66

We can simply add a new reason string.

Reply:
Nice one, I did as suggested. I didn't know about that metric.